prometheus-json-exporter: unstable-2017-10-06 -> 0.2.0

WilliButz opened the pull request:
###### Motivation for this change
#107587

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

fixes #107587

WilliButz committed:
prometheus-json-exporter: unstable-2017-10-06 -> 0.2.0

WilliButz committed:
nixos/prometheus-json-exporter: update modules & tests, add release notes

lukegb commented on nixos/tests/prometheus-exporters.nix:
```diff
@@ -241,7 +242,9 @@ let
         wait_for_open_port(7979)
-        succeed("curl -sSf localhost:7979/metrics | grep -q 'json_test_metric 1'")
+        succeed(
+            "curl -sSf 'localhost:7979/metrics?target=http://localhost' | grep -q 'json_test_metric 1'"
```
This test doesn't seem to pass: https://github.com/NixOS/nixpkgs/pull/108179/checks?check_run_id=1634971193

```
Exception: command `curl -sSf 'localhost:7979/metrics?target=http://localhost' | grep -q 'json_test_metric 1'` failed (exit code 1)
```

WilliButz replied:
Oh yeah, I missed switching to the probe endpoint. Thanks for the hint :+1:

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108179` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>
<summary>
1 package built:
</summary>
<ul>
<li>prometheus-json-exporter</li>
</ul>
</details>

Ma27 committed:
nixos/prometheus-exporters: minor doc improvements

* Content of `programlisting` shouldn't be indented, otherwise it's
  weirdly indented in the output.
* Use `<xref linkend=.../>` in the release notes: then users can
  directly go to the option documentation when reading release notes.
* Don't use docbook tags in `mkRemovedOptionModule`: it's only used
  during evaluation where docbook isn't rendered.
